### PR TITLE
Replace LinkedAccount in OCEX pallet with the one in polkadex-sgx-primitives

### PR DESCRIPTION
**Pull request by arrudagates**
This PR requires that the PR https://github.com/Polkadex-Substrate/polkadexTEE-worker/pull/188 in polkadexTEE-worker be merged to develop

### CODE REVIEW

**Comment by arrudagates** [comment]
Depends on https://github.com/Polkadex-Substrate/polkadexTEE-worker/pull/207 being merged.

File: pallets/ocex/src/lib.rs [hunk 1]
```diff
@@ -116,13 +90,13 @@ decl_error! {
...
         LastAccount get(fn key) config(): T::AccountId;
-        pub MainAccounts get(fn get_main_accounts): map hasher(blake2_128_concat) T::AccountId => LinkedAccount<T>;
+        pub MainAccounts get(fn get_main_accounts): map hasher(blake2_128_concat) T::AccountId => LinkedAccount;
     }
     add_extra_genesis {
         config(genesis_account): T::AccountId;
         build( |config: &GenesisConfig<T>| {
 
-            let linked_account_object = LinkedAccount::from(config.genesis_account.clone(), config.genesis_account.clone());
+            let linked_account_object = LinkedAccount::from(AccountId::new(config.genesis_account.clone().encode().as_slice().try_into().unwrap()), AccountId::new(config.genesis_account.clone().encode().as_slice().try_into().unwrap()));
```
Comment (arrudagates) [inline pallets/ocex/src/lib.rs:99]: This is not right, the sizes of do not match, therefore it fails to make the array as it expects `[u8; 32]`

Comment (arrudagates) [reply pallets/ocex/src/lib.rs:99]: Resolved

Comment (Gauthamastro) [reply pallets/ocex/src/lib.rs:99]: Never use unwrap in Substrate Runtime.

@zktony Please provide a better solution for creating LinkedAccount struct.

File: pallets/ocex/src/lib.rs [hunk 2]
```diff
@@ -239,9 +213,11 @@ decl_module! {
 impl<T: Config> Module<T> {
     // Note add_proxy doesn't check if given main or proxy is already registered
     pub fn add_proxy_(main: T::AccountId, proxy: T::AccountId) -> Result<(), Error<T>> {
-        let mut acc: LinkedAccount<T> = <MainAccounts<T>>::get(&main);
+        let mut acc: LinkedAccount = <MainAccounts<T>>::get(&main);
         if acc.proxies.len() < T::ProxyLimit::get() {
-            acc.proxies.push(proxy);
+            acc.proxies.push(AccountId::new(
+                proxy.encode().as_slice().try_into().unwrap(),
```
Comment (Gauthamastro) [inline pallets/ocex/src/lib.rs:219]: Same here

File: pallets/ocex/src/lib.rs [hunk 3]
```diff
@@ -253,11 +229,13 @@ impl<T: Config> Module<T> {
...
         <MainAccounts<T>>::try_mutate(
             main.clone(),
-            |ref mut linked_account: &mut LinkedAccount<T>| {
+            |ref mut linked_account: &mut LinkedAccount| {
                 let index = linked_account
                     .proxies
                     .iter()
-                    .position(|x| *x == proxy)
+                    .position(|x| {
+                        *x == AccountId::new(proxy.encode().as_slice().try_into().unwrap())
```
Comment (Gauthamastro) [inline pallets/ocex/src/lib.rs:237]: ^^

File: pallets/ocex/src/lib.rs [hunk 4]
```diff
@@ -276,11 +254,15 @@ impl<T: Config> Module<T> {
     pub fn register_acc(sender: T::AccountId) -> Result<(), Error<T>> {
         let last_account: T::AccountId = <LastAccount<T>>::get();
         <MainAccounts<T>>::try_mutate(last_account.clone(), |ref mut last_linked_account| {
-            let new_linked_account: LinkedAccount<T> =
-                LinkedAccount::from(last_account, sender.clone());
+            let new_linked_account: LinkedAccount = LinkedAccount::from(
+                AccountId::new(last_account.encode().as_slice().try_into().unwrap()),
+                AccountId::new(sender.clone().encode().as_slice().try_into().unwrap()),
```
Comment (Gauthamastro) [inline pallets/ocex/src/lib.rs:259]: ^^

File: pallets/ocex/src/lib.rs [hunk 5]
```diff
@@ -276,11 +254,15 @@ impl<T: Config> Module<T> {
...
+            );
             <MainAccounts<T>>::insert(&sender, new_linked_account);
             <LastAccount<T>>::put(&sender);
-            last_linked_account.next = Some(sender);
+            last_linked_account.next = Some(AccountId::new(
+                sender.encode().as_slice().try_into().unwrap(),
```
Comment (Gauthamastro) [inline pallets/ocex/src/lib.rs:264]: ^^

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**